src/prefect/engine/results/s3_result.py

```diff
@@ -167,7 +167,7 @@ def exists(self, location: str, **kwargs: Any) -> bool:
                 Bucket=self.bucket, Key=location.format(**kwargs)
             ).load()
         except botocore.exceptions.ClientError as exc:
-            if exc.response["Error"]["Code"] == "404":
+            if exc.response["Error"]["Code"] == "NoSuchKey":
```

Comment:
Are we confident this is universally the error code we'll see?  E.g., are there some configurations that might return `404`s?

Reply:
I don't fully recall where this came from (besides some old github issues that look to be dated) but it appears as the API now returns worded error responses as opposed to status codes https://docs.aws.amazon.com/AmazonS3/latest/API/ErrorResponses.html#ErrorCodeList